Rakefile

```diff
@@ -17,3 +17,160 @@ namespace :pkg do
+      opts.on('-h', '--help', 'Prints this help') { puts opts; exit }
+    end.parse!
+
+    plugin_name = "hammer_cli_foreman_#{options[:name]}"
+    plugin_path = File.expand_path(plugin_name.tr('_', '-'), options[:path])
+
+    FileUtils.mkpath("#{plugin_path}/lib/#{plugin_name}")
+    FileUtils.mkpath("#{plugin_path}/config")
+    FileUtils.mkpath("#{plugin_path}/test")
+    FileUtils.mkpath("#{plugin_path}/locale")
+    FileUtils.cd(plugin_path) do
+      capitalized_bits = options[:name].split('_').map(&:capitalize)
+      plugin_namespace = "HammerCLIForeman#{capitalized_bits.join}"
+
+      File.open("config/foreman_#{options[:name]}.yml", 'w') do |config|
+        config.write <<-EOF
+:foreman_#{options[:name]}:
+  :enable_module: true
+EOF
+      end
+      FileUtils.cp(File.expand_path('LICENSE', __dir__), FileUtils.pwd)
+      File.open('README.md', 'w') do |readme|
+        readme.write <<-EOF
+# TODO
+EOF
+      end
+      File.open('Gemfile', 'w') do |gemfile|
+        gemfile.write <<-EOF
+source "https://rubygems.org"
+
+gemspec
+EOF
+      end
+      File.open("#{plugin_name}.gemspec", 'w') do |gemspec|
+        gemspec.write <<-EOF
+lib = File.expand_path('../lib', __FILE__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require '#{plugin_name}/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = '#{plugin_name}'
+  spec.version       = #{plugin_namespace}.version.dup
+  spec.authors       = ['#{options[:author]}']
+  spec.email         = ['#{options[:email]}']
+  spec.homepage      = 'https://github.com/theforeman/#{plugin_name.tr('_', '-')}'
+  spec.license       = 'GPL-3.0'
+
+  spec.platform      = Gem::Platform::RUBY
+  spec.summary       = 'Foreman #{capitalized_bits.join(' ')} plugin for Hammer CLI'
+
+  # TODO: Don't forget to update required files accordingly!
+  spec.files         = Dir['{lib,config}/**/*', 'LICENSE', 'README*']
+  spec.require_paths = ['lib']
+  spec.test_files    = Dir['{test}/**/*']
+
+  spec.add_dependency 'hammer_cli_foreman', '>= 2.0.0'
+end
+EOF
+      end
+      File.open("lib/#{plugin_name}/version.rb", 'w') do |version|
+        version.write <<-EOF
+module #{plugin_namespace}
+  def self.version
+    @version ||= Gem::Version.new '0.0.1'
+  end
+end
+EOF
+      end
+      File.open("lib/#{plugin_name}.rb", 'w') do |plugin|
+        plugin.write <<-EOF
+module #{plugin_namespace}
+  require 'hammer_cli'
+  require 'hammer_cli_foreman'
+
+  require '#{plugin_name}/version'
+  require '#{plugin_name}/todo'
+
+  HammerCLI::MainCommand.lazy_subcommand(
+    'todo',
+    'Manage TODO',
+    '#{plugin_namespace}::TodoCommand',
+    '#{plugin_name}/todo'
+  )
+end
+EOF
+      end
+      File.open("lib/#{plugin_name}/todo.rb", 'w') do |todo_command|
+        todo_command.write <<-EOF
+module #{plugin_namespace}
+  class TodoCommand < HammerCLIForeman::Command
```

Comment:
I'd replace `Todo` with a more conventional noun to make it obvious that this is a resource/object/entity to be managed. And now for the hard part - to find a name for the example resource. Let the bikeshedding begin! :grin:

Reply:
:) Well, I've decided to use general `Resource`, but I'm still open for suggestions :D